### PR TITLE
feat(sql-udf): support named sql udf

### PR DESCRIPTION
**Pull request by xzhseh**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Related #10151.

As titled, this PR supports named sql udf.

The following definitions are valid now.

- normal definitions
```sql
create function add_named(a INT, b INT) returns int language sql as 'select a + b';
```

```sql
create function sub_named(a INT, b INT) returns int language sql as 'select a - b';
```

- mixed parameter with named / anonymous parameters
```sql
create function add_sub_mix(INT, a INT, INT) returns int language sql as 'select $1 + a + $3';
```

- valid, will not be rejected by semantic check
```sql
create function corner_case(INT, a INT, INT) returns varchar language sql as $$select '$1 + a + $3'$$;
```

For detailed use cases and potential error handling cases, check the test cases.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [x] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

Support named sql udf, see the above description for details.

</details>

### CODE REVIEW

File: src/frontend/src/handler/create_sql_function.rs [hunk 1]
```diff
@@ -33,14 +33,24 @@ use crate::expr::{ExprImpl, Literal};
...
-            (format!("${i}"), mock_expr.clone())
+            (format!("${i}"), mock_expr)
         })
-        .collect()
+        .collect();
+
+    for (i, arg_name) in arg_names.into_iter().enumerate() {
+        let mock_expr = ExprImpl::Literal(Box::new(Literal::new(None, arg_types[i].clone())));
+        ret.insert(arg_name, mock_expr);
+    }
```
Comment (wangrunji0408) [inline src/frontend/src/handler/create_sql_function.rs:51]: Need to skip when `arg_name` is empty?

**Review by wangrunji0408 — APPROVED** [review]
LGTM! Thanks!

File: e2e_test/udf/sql_udf.slt
[the remark this answers is not in the document]
Comment (xzhseh) [reply e2e_test/udf/sql_udf.slt:1]: Do you mind me putting this test case refactor to a subsequent PR? 🥵

File: src/frontend/src/handler/create_sql_function.rs (hunk 1 above)
Comment (xzhseh) [reply src/frontend/src/handler/create_sql_function.rs:51]: Actually no need to skip here, the empty key in `UdfContext` is totally fine and will not affect semantic check. 😄

File: e2e_test/udf/sql_udf.slt [hunk 2]
```diff
@@ -13,6 +13,31 @@ create function sub(INT, INT) returns int language sql as 'select $1 - $2';
...
+statement ok
+create function add_named(a INT, b INT) returns int language sql as 'select a + b';
+
+# Create another named sql udf
+statement ok
+create function sub_named(a INT, b INT) returns int language sql as 'select a - b';
+
+# Mixed parameter with named / anonymous parameters
+statement ok
+create function add_sub_mix(INT, a INT, INT) returns int language sql as 'select $1 + a + $3';
```
[the remark this answers is not in the document]
Comment (xzhseh) [reply e2e_test/udf/sql_udf.slt:26]: Oops, I'll adjust this later. 🤣

File: e2e_test/udf/sql_udf.slt
[the remark this answers is not in the document]
Comment (wangrunji0408) [reply e2e_test/udf/sql_udf.slt:1]: Feel free to do it anytime. 🫡

**Pull request merged**